.evergreen/pkgconfig-tests.sh

```diff
@@ -38,6 +38,12 @@ else
     chmod u+x ./.evergreen/find-cmake.sh
     # Amazon Linux 2 (arm64) has a very old system CMake we want to ignore
     IGNORE_SYSTEM_CMAKE=1 . ./.evergreen/find-cmake.sh
+    # Check if on macOS with arm64.
+    OS_NAME=$(uname -s | tr '[:upper:]' '[:lower:]')
+    MARCH=$(uname -m | tr '[:upper:]' '[:lower:]')
+    if [ "darwin" = "$OS_NAME" -a "arm64" = "$MARCH" ]; then
+        ADDITIONAL_CMAKE_FLAGS="$ADDITIONAL_CMAKE_FLAGS -DCMAKE_OSX_ARCHITECTURES=arm64"
+    fi
```

Comment:
Similar to the changes in CDRIVER-4240, this resolves an error with libbson being built for x86_64: It looks due to a mismatch in architectures:

```
+ gcc -DMONGOCRYPT_STATIC_DEFINE -DKMS_MSG_STATIC -DBSON_STATIC -I/Users/dev-user/kevinAlbs/libmongocrypt/pkgconfig_tests/install/include/libbson-1.0 -I/Users/dev-user/kevinAlbs/libmongocrypt/pkgconfig_tests/install/libmongocrypt/include/mongocrypt -o example-state-machine test/example-state-machine.c /Users/dev-user/kevinAlbs/libmongocrypt/pkgconfig_tests/install/libmongocrypt/lib/libmongocrypt-static.a /Users/dev-user/kevinAlbs/libmongocrypt/pkgconfig_tests/install/libmongocrypt/lib/libkms_message-static.a /Users/dev-user/kevinAlbs/libmongocrypt/pkgconfig_tests/install/libmongocrypt/lib/libbson-static-1.0.a -pthread -framework CoreFoundation -framework Security
ld: warning: ignoring file /Users/dev-user/kevinAlbs/libmongocrypt/pkgconfig_tests/install/libmongocrypt/lib/libbson-static-1.0.a, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file /Users/dev-user/kevinAlbs/libmongocrypt/pkgconfig_tests/install/libmongocrypt/lib/libkms_message-static.a, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file /Users/dev-user/kevinAlbs/libmongocrypt/pkgconfig_tests/install/libmongocrypt/lib/libmongocrypt-static.a, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
Undefined symbols for architecture arm64:
  "_bcon_new", referenced from:
      _main in example-state-machine-d843ff.o
```